status/status.go

```diff
@@ -148,60 +133,81 @@ OuterLoop:
 						resp.Body.Close()
 					}
 				}
-
-				// Mattermost
-				if configuration.MattermostServer != "" {
-					if err := UpdateMattermost(nc, status); err != nil {
-						log.Printf("Mattermost error: %s\n", err)
-					} else {
-						log.Printf("Mattermost updated")
-					}
-				}
 			}
 			first = false
 			time.Sleep(time.Second)
 		}
 	}
 }
 
-func UpdateMattermost(nc *http.Client, status bool) error {
-	mm := model.NewAPIv4Client(configuration.MattermostServer)
-	mm.HttpClient = nc
-	mm.SetToken(configuration.MattermostToken)
-
-	channel, resp := mm.GetChannel(configuration.MattermostChannelId, "")
-	if channel == nil {
-		return fmt.Errorf("Get channel: %+v", resp)
+// UpdateTopic modifies the topic (IRC, Mattermost) by matching `re` and replacing
+// the subexpression by `new`. The regexp must have exactly one subexpression.
+func (ss *SWITCHSTATE) UpdateTopic(irccon *irc.Connection, nc *http.Client, re *regexp.Regexp, new string) {
+	err := ss.updateTopicIRC(irccon, re, new)
+	if err != nil {
+		log.Printf("updateTopicIRC error: %s\n", err)
 	}
 
-	const labOpen = "|| LAB OPEN ||"
-	const labClosed = "|| LAB CLOSED ||"
-
-	var strStatus string
-	if status {
-		strStatus = labOpen
-	} else {
-		strStatus = labClosed
+	if configuration.MattermostServer != "" {
```

Comment:
Is IRC server never disconnected either?

Reply:
Yeah the whole thing is built around the IRC connection: https://github.com/FOULAB/foubot2/blob/5f082f1adc2c8bc1e2c6c257e9347e7cd952bc24/main.go#L113

This code doesn't even run if IRC is not connected.

It's not ideal (in theory we would want foubot to work when IRC is down but Mattermost is up) but 🤷‍♂️ for now.